ci.md

```diff
@@ -0,0 +1,229 @@
+🟡 state from a couple of minutes up to an entire day. If a build is in this
+state for too long, there may be some issue with infrastructure and the
+build didn't run or didn't report back.
+
+Failing builds ❌ usually require more investigation:
+
+* If the build is `Required`, something is definitely wrong and must be fixed
+  before merging the pull request.
+* If the build is not required, investigation is needed to find out if the
+  failure is pre-existing or if it is being introduced by the pull request.
+
+Clicking on the `Details` button of a failing build takes you to the build so
+that you can inspect it.
+
+Builds can fail for a variety of reasons, for example:
+
+* The code **failed to compile**. This should never be accepted, even in non-required
+  checks, and must be fixed.
+    * **Jenkins**: When there's a compilation failure, the build is marked red 🔴.
+                   On the build's page, click on `Console Output` then scroll until
+                   you find the compilation errors under the `compiling` section.
+    * **Actions**: When there's a compilation failure, the build is marked red ❌.
+                   On the build logs, the compilation failure should be under the
+                   `make` collapsible.
+* There are **test failures**. Test failures that aren't pre-existing must be fixed.
+    * **Jenkins**: When there are test failures, the build is marked yellow 🟡.
+                   On the build's page, the failing tests are listed under
+                   `Test Result`. Click on a test's name, then on "History" on the
+                   left to see if the test has failed before.
+                   It's often helpful to see the full test logs on `Console Output`.
+    * **Actions**: When there are test failures, the build is marked red ❌.
+                   On the build logs, the test failures should be under the
+                   `make test` collapsible.
+    * Check which tests are known failures by searching for the test name on
+      issues.
+* There are **warnings**. Warnings that aren't pre-existing must be fixed.
+    * **Jenkins**: When there are warnings, the build is marked yellow 🟡.
+                   On the build's page, the warnings are listed under
+                   `GNU C Compiler` and `CMake`. You can also check the
+                   `Console Output` for the full warning logs.
+    * **Actions**: Does not detect compiler or CMake warnings.
+* There are **static checker failures**. All code checker failures must be fixed.
+    * **Jenkins**: Does not run static checkers.
+    * **Actions**: When there are code check failures, the build is marked red ❌.
+                   On the build logs, the checker failures should be under the
+                   `Code check` collapsible.
+* There are **infrastructure failures**. Contributors should leave a comment to
+    maintainers, who willl report these to the build farmer.
```

Comment:
```suggestion
* There are **infrastructure failures**. Contributors should leave a comment to
    maintainers (this can be done by leaving a comment on the PR), who will report these to the build farmer.
```